common/script/aria.js

```diff
@@ -7,746 +20,395 @@
  * localprops: local properties and states
  */
 
-var roleInfo = {};
+const roleInfo = {};
 
-function ariaAttributeReferences() {
-    var propList = {};
-    var globalSP = [];
+/**
+ * Populates propList for given sdef/pdef
+ * @param {Object} propList -
+ * @param {HTMLElement} item - from nodeList.forEach
+ */
+const populatePropList = function (propList, item) {
+    const type = item.localName === "pdef" ? "property" : "state";
+    const content = item.innerHTML;
+    const title = item.getAttribute("title") || content;
+    const dRef = item.nextElementSibling;
+    const desc = cloneWithoutIds(dRef.firstElementChild).innerHTML;
+    propList[title] = {
+        is: type,
+        title: title,
+        name: content,
+        desc: desc,
+        roles: [],
+    };
+};
 
-    var skipIndex = 0;
-    var myURL = document.URL;
-    if (myURL.match(/\?fast/)) {
-        skipIndex = 1;
+/**
+ * Populates globalSP for given sdef/pdef
+ * @param {Object} propList -
+ * @param {Object} globalSP -
+ * @param {HTMLElement} item - from nodeList.forEach
+ */
+const populateGlobalSP = function (propList, globalSP, item) {
+    const title = item.getAttribute("title") || item.innerHTML;
+    const container = item.parentElement;
+    const itemEntry = propList[title];
+
+    const applicabilityText = container.querySelector(
+        "." + itemEntry.is + "-applicability"
+    ).innerText;
+    const isDefault = applicabilityText === "All elements of the base markup";
+    const isProhibited =
+        applicabilityText ===
+        "All elements of the base markup except for some roles or elements that prohibit its use";
+    const isDeprecated =
+        applicabilityText === "Use as a global deprecated in ARIA 1.2";
```

Comment:
From https://github.com/w3c/aria-common/pull/106#discussion_r1462542533

> I see, this is awful. Definitely, it makes more sense to start with a JSON object rather than looking for this text..